borg/testsuite/hashindex.py

```diff
@@ -100,3 +101,94 @@ def test_chunkindex_merge(self):
         assert idx1[H(2)] == (7, 200, 200)
         assert idx1[H(3)] == (3, 300, 300)
         assert idx1[H(4)] == (6, 400, 400)
+
+    def test_chunkindex_summarize(self):
+        idx = ChunkIndex()
+        idx[H(1)] = 1, 1000, 100
+        idx[H(2)] = 2, 2000, 200
+        idx[H(3)] = 3, 3000, 300
+
+        size, csize, unique_size, unique_csize, unique_chunks, chunks = idx.summarize()
+        assert size == 1000 + 2 * 2000 + 3 * 3000
+        assert csize == 100 + 2 * 200 + 3 * 300
+        assert unique_size == 1000 + 2000 + 3000
+        assert unique_csize == 100 + 200 + 300
+        assert chunks == 1 + 2 + 3
+        assert unique_chunks == 3
+
+
+class HashIndexRefcountingTestCase(BaseTestCase):
+    MAX_REF = 2**31-1
+
+    def test_chunkindex_overflow(self):
+        idx = ChunkIndex()
+        idx[H(1)] = self.MAX_REF - 1, 1, 2
+
+        # 5 is arbitray, any number of incref/decrefs shouldn't move it once it maxes out
+        for i in range(5):
+            refcount, *_ = idx.incref(H(1))
+            assert refcount == self.MAX_REF
+        for i in range(5):
+            refcount, *_ = idx.decref(H(1))
+            assert refcount == self.MAX_REF
+
+    def test_chunkindex_merge_overflow1(self):
+        # Check that it does not overflow at 2**31-2
+        idx1 = ChunkIndex()
+        # n.b. 2**31-1 is odd
+        half = self.MAX_REF // 2
```

Comment:
that is 2^30 - 1.

2 * that == 2^31 - 2.

so, why should that "overflow" (see comment)?